### PR TITLE
[nrf noup] Adapt Android CHIPTool guide to Wi-Fi devices

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Switch the reference example in the guide from the lock to the light bulb since there is no dedicated Door Lock screen in the application.
Make the guide a little bit more aware of the world of Wi-Fi devices.

### CODE REVIEW

File: docs/guides/nrfconnect_android_commissioning.md [hunk 1]
```diff
@@ -60,27 +61,33 @@ applications:
...
-        replace this DK with another compatible device, such as the nRF5340 DK.
-
--   Smartphone compatible with Android 8.0 or later
--   Raspberry Pi Model 3B+ or newer (along with an SD card with at least 8 GB of
-    memory)
--   Wi-Fi Access Point supporting IPv6 (without the IPv6 Router Advertisement
+-   1x smartphone with Android 8+
+-   1x Wi-Fi Access Point supporting IPv6 (without the IPv6 Router Advertisement
     Guard enabled on the router)
+-   1x nRF52840 DK (PCA10056) for running the example application. You can
```
Comment (kkasperczyk-no) [inline docs/guides/nrfconnect_android_commissioning.md:67]: Can we clarify here which DKs are compatible with Thread and Wi-Fi?

File: docs/guides/nrfconnect_android_commissioning.md [hunk 2]
```diff
@@ -164,24 +171,25 @@ To prepare the accessory device for commissioning, complete the following steps:
...
+1.  Enable **Bluetooth** and **Location** services on your smartphone.
+2.  Connect the smartphone to your Wi-Fi Access Point. If you're testing a
+    Thread device, verify that the OpenThread Border Router is connected to the
+    same AP.
+3.  Open the CHIPTool application on your smartphone.
+4.  Tap the **PROVISION CHIP DEVICE WITH THREAD** button and scan the
+    commissioning QR code.
+    > If you're testing a Wi-Fi device, you need to tap the
+    > **PROVISION CHIP DEVICE WITH WI-FI** button, instead.
+    The network credentials screen appears.
```
Comment (greg-fer) [inline docs/guides/nrfconnect_android_commissioning.md:186]:
```suggestion
4.  Depending on your testing scenario, tap one of the following buttons and scan the
    commissioning QR code:

    - **PROVISION CHIP DEVICE WITH THREAD** for Matter over Thread.
    - **PROVISION CHIP DEVICE WITH WI-FI** for Matter over Wi-Fi.

    The network credentials screen appears.
```

File: docs/guides/nrfconnect_android_commissioning.md [hunk 3]
```diff
@@ -164,24 +171,25 @@ To prepare the accessory device for commissioning, complete the following steps:
...
+5.  In the network credentials screen, specify parameters of your Thread or
+    Wi-Fi network and tap the **SAVE NETWORK** button. Several notifications
+    appear, informing you of the progress of scanning, connecting, and pairing
+    with the device. At the end of this process, the application returns to the
+    main menu.
```
Comment (greg-fer) [inline docs/guides/nrfconnect_android_commissioning.md:191]:
```suggestion
5.  In the network credentials screen, specify parameters of your network and tap the **SAVE NETWORK** button. Several notifications
    appear, informing you of the progress of scanning, connecting, and pairing
    with the device. At the end of this process, the application returns to the
    main menu.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: docs/guides/nrfconnect_android_commissioning.md [hunk 4]
```diff
@@ -164,24 +171,25 @@ To prepare the accessory device for commissioning, complete the following steps:
...
-
-You will see the "Network provisioning completed" message when the accessory
-device successfully joins the Thread network.
+To commission the accessory device into the Matter fabric, complete the
+following steps:
+
+1.  Enable **Bluetooth** and **Location** services on your smartphone.
+2.  Connect the smartphone to your Wi-Fi Access Point. If you're testing a
+    Thread device, verify that the OpenThread Border Router is connected to the
+    same AP.
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:180]: No, this should already be done as a part of the OTBR setup (covered by external docs). I can move this advise to the section which references external OTBR docs though.

File: docs/guides/nrfconnect_android_commissioning.md [hunk 5]
```diff
@@ -60,27 +61,33 @@ applications:
...
+
 
 <hr>
 
 <a name="setting-up-thread-border-router"></a>
 
 ## Setting up Thread Border Router
 
+> This step is only needed if you're testing a Thread device and can be skipped
+> if the device operates in a Wi-Fi network.
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:89]: It's just unnecessary, I'll follow your suggestion.

File: docs/guides/nrfconnect_android_commissioning.md (hunk 3 above)
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:191]: Done

File: docs/guides/nrfconnect_android_commissioning.md (hunk 2 above)
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:186]: Done

File: docs/guides/nrfconnect_android_commissioning.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:180]: Move to another section

File: docs/guides/nrfconnect_android_commissioning.md [hunk 6]
```diff
@@ -60,27 +61,33 @@ applications:
...
+-   1x nRF52840 DK for running the
+    [OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor)
+    firmware. You can replace this DK with another compatible device, such
+    as the nRF52840 Dongle.
+    > This piece of hardware is only needed if you're testing a Thread device
+    > and can be skipped if the device operates in a Wi-Fi network.
+-   1x Raspberry Pi Model 3B+ or newer (along with an SD card with at least 8 GB of
+    memory)
+    > This piece of hardware is only needed if you're testing a Thread device
+    > and can be skipped if the device operates in a Wi-Fi network.
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:79]: Sure

File: docs/guides/nrfconnect_android_commissioning.md (hunk 1 above)
Comment (Damian-Nordic) [reply docs/guides/nrfconnect_android_commissioning.md:67]: Added clarification.

[the remark this answers is not in the document]
Comment (greg-fer) [reply docs/guides/nrfconnect_android_commissioning.md:67]: No, the NCS docs should rather have fully-spelled numbers. In the Matter docs, we can be more liberal. Leave as-is.

**Review by greg-fer — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**